app/scripts/modules/azure/serverGroup/configure/serverGroupConfiguration.service.spec.js

```diff
@@ -164,5 +164,33 @@ describe('Service: azureServerGroupConfiguration', function() {
       expect(this.command.backingData.filtered.securityGroups).toEqual([]);
       expect(this.command.viewState.securityGroupConfigured).toBeFalse;
     });
+
+    it('returns empty zone list if region is not supported', function() {
+      this.command.region = 'eastasia';
+      this.command.backingData.credentialsKeyedByAccount = {};
+      this.command.backingData.credentialsKeyedByAccount[this.command.credentials] = {
+        regionsSupportZones: [],
+        availabilityZones: ['1', '2', '3'],
+      };
+
+      service.configureZones(this.command);
+
+      expect(this.command.backingData.filtered.zones).toEqual([]);
+    });
+
+    it('returns actual zone list if region not supported', function() {
+      this.command.region = 'eastasia';
+      this.command.backingData.credentialsKeyedByAccount = {};
+      this.command.backingData.credentialsKeyedByAccount[this.command.credentials] = {
+        regionsSupportZones: ['eastasia'],
+        availabilityZones: ['1', '2', '3'],
+      };
+
+      service.configureZones(this.command);
+
+      expect(this.command.backingData.filtered.zones).toEqual(
+        this.command.backingData.credentialsKeyedByAccount[this.command.credentials].availabilityZones,
+      );
+    });
```

Comment:
I suggest adding a test for behaviour expected when `this.command.region` is null or undefined.